src/transformers/processing_utils.py

```diff
@@ -830,6 +827,60 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
             output_kwargs[modality].update(output_kwargs["common_kwargs"])
         return output_kwargs
 
+    def _check_reversed_images_text(self, images, text):
+        """
+        For backward compatibility: reverse the order of `images` and `text` inputs if they are swapped.
+        This method should only be called for processors where `images` and `text` have been swapped for uniformization purposes.
+        Note that this method assumes that two `None` inputs are valid inputs. If this is not the case, it should be handled
+        in the processor's `__call__` method before calling this method.
+        """
+
+        def _is_valid_text_input(t):
+            if isinstance(t, str):
+                # Strings are fine
+                return True
+            elif isinstance(t, (list, tuple)):
+                # List are fine as long as they are...
+                if len(t) == 0:
+                    # ... empty
+                    return True
+                elif isinstance(t[0], str):
+                    # ... list of strings
+                    return True
+                elif isinstance(t[0], (list, tuple)):
+                    # ... list with an empty list or with a list of strings
+                    return len(t[0]) == 0 or isinstance(t[0][0], str)
+                else:
```

Comment:
i don't think we work with nested list so texts in processors, that's usually for tokenizers with text-pair. Also the empty list, is it a valid input type?

What can be a valid type is a encoded text, which is missing here

Reply:
I took an existing `_is_valid_text_input` function that is defined in several tokenizers, but you are right it should probably be adapted a bit for processors.
Encoded text are lists of int right? Also it seems we usually don't advertise `EncodedInput` as an acceptable input type for text in processor, so I don't know if that is in purpose or not.

Reply:
we can also reuse the `is_valid_image` util

Reply:
Yep, I'm using `valid_images `which recursively used `is_valid_image` to check for nested images